Prefix all schnorr tags with `SCHNORR`

tessico opened the pull request:
closes #7

tessico committed:
Prefix all schnorr tags with `SCHNORR`

tessico commented:
Also @EspressoSystems/jellyfish

alxiong commented on src/lib.rs:
```diff
@@ -109,6 +108,9 @@ pub mod jellyfish {
         pub const BLSSIGNINGKEY: &str = "BLSSIGNINGKEY";
         pub const BLSSIG: &str = "BLSSIG";
 
+        pub const SCHNORRSIGNKEYPAIR: &str = "SCHNORRSIGNKEYPAIR";
         pub const SCHNORRVERKEY: &str = "SCHNORRVERKEY";
+        pub const SCHNORRSIGNINGKEY: &str = "SCHNORRSIGNINGKEY";
+        pub const SCHNORRSIG: &str = "SCHNORRSIG";
```
can we use snake case, this is hard to read.

tessico replied:
Since we're already making backwards incompatible changes, yes why not!

tessico replied:
verygoodidea

alxiong replied:
sry, this is really minor, but I think we should always delimit between two words, such as `SCHNORR_SIGNING_KEY` instead of `SCHNORR_SIGNINGKEY`

tessico replied:
@alxiong Since we're at it, why don't we rename to the notation used in [the draft](https://datatracker.ietf.org/doc/draft-irtf-cfrg-bls-signature/)? Some [libs](https://github.com/arkworks-rs/crypto-primitives/blob/bd447455510878adf82894cb23766dc914c31c32/src/signature/mod.rs) seem to follow the same notation, while [others](https://github.com/RustCrypto/signatures) use mixed names (e.g. SecretKey -> SigningKey).
That would, of course, mean that we'd need to rename the types on our [`SignatureScheme`](https://github.com/EspressoSystems/jellyfish/blob/9bc08d2e5e943ea2610eb02064263ceb38478280/primitives/src/signatures/mod.rs#L15) trait in jellyfish.

alxiong replied:
Generally, I don't think we need to follow IRTF draft for naming convention. 

But on the topic of "SigningKey" v.s. `SecretKey`, I don't have a preference. [Boneh's book](https://toc.cryptobook.us/book.pdf#chapter.13) says "signing", [wiki ](https://en.wikipedia.org/wiki/Digital_signature)and many standard use the more general "secret".

by least disruption principle, we probably should keep the associated type names in `trait SignatureScheme` unchanged.

tessico committed:
remove SIG tag, replaced by `SCHNORR_SIG`

tessico commented on src/lib.rs:
```diff
@@ -98,7 +98,6 @@ pub mod jellyfish {
         pub const VERKEY: &str = "VERKEY";
-        pub const SIGNKEYPAIR: &str = "SIGNKEYPAIR";
         pub const SIG: &str = "SIG";
         pub const NODE: &str = "NODE";
```
Also, as far as I can see, the `NODE` and `LEAF` tags are not used. Should we remove them?

tessico replied:
I've also removed `SIG`, since it's now replaced by `SCHNORR_SIG`.

alxiong replied:
just curious, how are you able to check their usage throughout all dependents?

tessico replied:
Since it would be defined in the jellyfish lib (if it were defined), if nothing is tagged there with these tags, then there can't be any dependency using it right?

tessico committed:
add underscores between signature type and data structure name

tessico committed:
Add underscore between each word

tessico committed:
remove VERKEY, replaced by SCHNORR_VER_KEY

tessico commented:
Latest changes:
- added underscores between each word
- renamed `...SIGNING_KEYPAIR` to `...KEY_PAIR`, as IMO the latter makes more sense

@alxiong WDYT?

alxiong commented on src/lib.rs:
```diff
@@ -97,18 +97,18 @@ pub mod hotshot {
-        pub const BLSSIGNINGKEY: &str = "BLSSIGNINGKEY";
-        pub const BLSSIG: &str = "BLSSIG";
+        pub const BLS_VER_KEY: &str = "BLS_VER_KEY";
+        pub const BLS_SIGNING_KEY: &str = "BLS_SIGNING_KEY_";
```
```suggestion
        pub const BLS_SIGNING_KEY: &str = "BLS_SIGNING_KEY";
```

alxiong commented on src/lib.rs:
```diff
@@ -97,18 +97,18 @@ pub mod hotshot {
+        pub const BLS_SIG: &str = "BLS_SIG";
 
-        pub const SCHNORRVERKEY: &str = "SCHNORRVERKEY";
+        pub const SCHNORR_KEY_PAIR: &str = "SCHNORR_KEY_PAIR";
+        pub const SCHNORR_VERK_EY: &str = "SCHNORR_VERKEY";
```
```suggestion
        pub const SCHNORR_KEY_PAIR: &str = "SCHNORR_KEY_PAIR";
        pub const SCHNORR_VER_KEY: &str = "SCHNORR_VER_KEY";
```

tessico replied:
Thanks @alxiong! That was weird...

tessico committed:
Apply suggestions from code review

Co-authored-by: Alex Xiong <alex.xiong.tech@gmail.com>
